src/api-v1.js

```diff
@@ -99,7 +99,15 @@ api.declare({
       // instead of a more complicated structure
 
       if (x === 0 && (result.status === 'absent' || result.status === 'error')) {
-        let validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
+        let validUrl;
+        try {
+          validUrl = await validateUrl(url, this.allowedPatterns, this.redirectLimit, this.ensureSSL, this.monitor);
```

Comment:
Jonas, we only do it when the item is not in the cache or had an error transferring and then, only once on the first attempt.  We do this so that a URL that is definitely invalid will be returned quickly.  Cached URLs don't follow the redirects. This is a fail-fast optimisation.  I'm on my phone so am being brief.